tests/test_topology.py

```diff
@@ -367,3 +370,72 @@ def test_component_idxs():
     # base topology w/host
     hgt_base = topology.HostGuestTopology(host_bps, bt)
     np.testing.assert_equal(hgt_base.get_component_idxs(), [solvent_idxs, mol_a_idxs + num_solvent_atoms])
+
+
+def test_relative_free_energy_forcefield():
+
+    with resources.path("timemachine.testsystems.data", "ligands_40.sdf") as path_to_ligand:
+        mol = next(Chem.SDMolSupplier(str(path_to_ligand), removeHs=False))
+
+    ff0 = Forcefield.load_from_file(DEFAULT_FF)
+    ff1 = Forcefield.load_from_file(DEFAULT_FF)
+
+    # Modify the charge parameters for ff1
+    ff1.q_handle.params += 1.0
+
+    fftop = topology.RelativeFreeEnergyForcefield(mol, ff0, ff1)
+    bt0 = topology.BaseTopology(mol, ff0)
+    bt1 = topology.BaseTopology(mol, ff1)
+
+    ordered_handles = ff0.get_ordered_handles()
+    bond_idx = ordered_handles.index(ff0.hb_handle)
+    angle_idx = ordered_handles.index(ff0.ha_handle)
+    proper_idx = ordered_handles.index(ff0.pt_handle)
+    improper_idx = ordered_handles.index(ff0.it_handle)
+    charge_idx = ordered_handles.index(ff0.q_handle)
+    lj_idx = ordered_handles.index(ff0.lj_handle)
+    ff0_params = ff0.get_ordered_params()
+    ff1_params = ff1.get_ordered_params()
+
+    combined_params = combine_ordered_params(ff0, ff1)
+    combined_qlj_params, combined_ubp = fftop.parameterize_nonbonded(
+        combined_params[charge_idx], combined_params[lj_idx]
+    )
+    qlj0_params, ubp0 = bt0.parameterize_nonbonded(ff0_params[charge_idx], ff0_params[lj_idx])
+    qlj1_params, ubp1 = bt1.parameterize_nonbonded(ff1_params[charge_idx], ff1_params[lj_idx])
+
+    coords = get_romol_conf(mol)
+    box = np.identity(3) * 99.0
+
+    def u_endstates(ubp, params) -> Tuple[float, float]:
+        u = ubp.bind(params).bound_impl(precision=np.float32)
+        _, _, u0_val = u.execute(coords, box, lam=0)
+        _, _, u1_val = u.execute(coords, box, lam=1)
+        return u0_val, u1_val
+
+    u0_combined, u1_combined = u_endstates(combined_ubp, combined_qlj_params)
+    u0, _ = u_endstates(ubp0, qlj0_params)
+    u1, _ = u_endstates(ubp1, qlj1_params)
```

Comment:
Not important since only executed once in test, but `ubp.bind(params).bound_impl(np.float32)` is an setup operation that only needs to be done once, so could consider 

`def get_impl(ubp, params): return ubp.bind(params).bound_impl(precision=np.float32); impl0 = get_impl(ubp0, qlj0_params); u0 = impl0.execute(coords, box, lam=0)` or similar. (In case the test is modified in the future to be called multiple times with different (coords, box, lam).)